src/components/sound.js

```diff
@@ -125,24 +127,48 @@ module.exports.Component = registerComponent('sound', {
       evt.detail.cameraEl.getObject3D('camera').add(listener);
     });
 
-    sound = this.sound = new THREE.PositionalAudio(listener);
-    el.setObject3D(this.attrName, sound);
-
-    sound.source.onended = function () {
-      sound.onEnded();
-      el.emit('sound-ended');
-    };
+    // Create [poolSize] audio instances and attach them to soundsPool
+    this.soundsPool = new THREE.Group();
+    for (var i = 0; i < this.data.poolSize; i++) {
+      var sound = new THREE.PositionalAudio(listener);
+      this.soundsPool.add(sound);
+    }
+    el.setObject3D(this.attrName, this.soundsPool);
 
-    return sound;
+    this.soundsPool.children.forEach(function (sound) {
+      sound.source.onended = function () {
+        sound.onEnded();
+        el.emit('sound-ended', {index: i});
+      };
+    });
   },
 
   playSound: function () {
-    if (!this.sound.source.buffer) { return; }
-    this.sound.play();
+    // Look for an unused sound in the pool and play it if found.
+    for (var i = 0; i < this.soundsPool.children.length; i++) {
```

Comment:
I used because I use `return` to exit the method if a instance is found on the pool, otherwise I show the warning message. I'll change it to use foreach and some boolean variable `isFound` that will determine if the message will be shown or not